### PR TITLE
Move Experiment Check Up Component Tree to Speed Up Load

### PR DESCRIPTION
**Pull request by dmcavoy**
Moved checks of experiment up the tree of components and out of components that are rendered on one page many times (just as bubbles).

I would love for someone to try this locally to make sure I didn't miss anything in my testing. 

Also could use some help showing that this does in fact improve performance. 

# Before (current staging locally)
~36 seconds to load
Screen Shot 2019-04-12 at 4 58 40 PM

# After (current branch locally)
~18 seconds to load
Screen Shot 2019-04-12 at 4 49 05 PM

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
LGTM!

File: apps/src/code-studio/components/progress/StageProgress.jsx [hunk 1]
```diff
@@ -61,6 +62,10 @@ class StageProgress extends Component {
     const {stageExtrasUrl, onStageExtras, stageTrophyEnabled} = this.props;
     let levels = this.props.levels;
 
+    const inMiniRubricExperiment = experiments.isEnabled(
```
Comment (maddiedierker) [inline apps/src/code-studio/components/progress/StageProgress.jsx:65]: i think this looks good as-is, but if performance is still an issue, i think we can remove checking the experiment in the `render()` method and instead set some state in `componentDidMount` like this:

```
componentDidMount() {
  this.state = {inMiniRubricExperiment: experiments.isEnabled(experiments.MINI_RUBRIC_2019)};
}

render() {
  const inMiniRubricExperiment = this.state.inMiniRubricExperiment;
}
```

there are a few places below where we could make this change as well.

**Review by maddiedierker — APPROVED** [review]
🎉 great job!

**Comment by davidsbailey** [comment]
LGTM once tests are passing! could run (including backticks)
```
`npm bin`/grunt storybookTest
``` 
to get a jump on some of the test failures if you don't want to wait for drone.

**Review by davidsbailey — APPROVED** [review]
🏇🏎 💨

**Comment by davidsbailey** [comment]
@dmcavoy I've disabled this test, which is a test-only problem affecting only the case where the experiment is enabled, to unblock this so we can get the fix deployed:
```
FAILED TESTS:
  unit tests
    progressReduxTest
      reducer
        statusForLevel
          ✖ returns LevelStatus.completed_assessment for assessment level when experiment is on
            PhantomJS 2.1.1 (Linux 0.0.0)
          expected 'perfect' to equal 'completed_assessment'

      + expected - actual

      -perfect
      +completed_assessment
```

**Pull request merged**